Doc/whatsnew/3.7.rst

```diff
@@ -881,6 +881,12 @@ Changes in the Python API
   recent to be more consistent with :mod:`traceback`.
   (Contributed by Jesse Bakker in :issue:`32121`.)
 
+* On Linux, the :attr:`socket.type <socket.socket.type>` attribute no
+  longer has :const:`socket.SOCK_NONBLOCK` or :const:`socket.SOCK_CLOEXEC`
+  flags applied, so checks like ``if sock.type == socket.SOCK_STREAM``
+  work as expected on all platforms.
```

Comment:
You may suggest to use getblocking() to check if the socket is non-blocking.

"On Linux" doesn't include FreeBSD. Maybe just remove "On Linux".